src/storage/db_interface_frontend.py

```diff
@@ -124,7 +124,7 @@ def all_uids_found_in_database(self, uid_list):
             return True
         query = self._build_search_query_for_uid_list(uid_list)
         number_of_results = self.get_firmware_number(query) + self.get_file_object_number(query)
-        return len(uid_list) == number_of_results
+        return number_of_results >= len(uid_list)
```

Comment:
Can you explain why this is necessary?

Reply:
The bug occurs if the same object is in the file object and firmware databases.
The number of uids is compared to the number of objects in the databases and (since one uid is in both databases) they don't match up.

Reply:
Originally, the plan was to fix this by merging both databases (if that ever happens remains to be seen)